client/ayon_equalizer/api/host.py

```diff
@@ -114,56 +171,146 @@ def get_context_data(self) -> dict:
             re.sub(
-                CONTEXT_REGEX,
-                f"AYON_CONTEXT::{update_str}::AYON_CONTEXT_END",
-                tde4.getProjectNotes()
+                AYON_METADATA_REGEX,
+                AYON_METADATA_GUARD.format(update_str),
+                tde4.getProjectNotes(),
             )
         )
         tde4.updateGUI()
 
-    def install(self):
+    def get_context_data(self) -> dict:
+        """Get context data from the current project."""
+        data = self.get_ayon_data()
+
+        return data.get(EQUALIZER_CONTEXT_KEY, {})
+
+    def update_context_data(self, data: dict, changes: dict) -> None:
+        """Update context data in the current project.
+
+        Args:
+            data (dict): Context data.
+            changes (dict): Changes to the context data.
+
+        Raises:
+            RuntimeError: If the context data is not found.
+
+        """
+        if not data:
+            return
+        ayon_data = self.get_ayon_data()
+        if EQUALIZER_CONTEXT_KEY not in ayon_data:
+            ayon_data[EQUALIZER_CONTEXT_KEY] = []
+
+        ayon_data[EQUALIZER_CONTEXT_KEY] = data
+        self.update_ayon_data(ayon_data)
+
+
+    def get_publish_instances(self) -> list[dict]:
+        """Get publish instances from the current project."""
+        data = self.get_ayon_data()
+        return data.get(EQUALIZER_INSTANCES_KEY, [])
+
+    def add_publish_instance(self, instance_data: dict) -> None:
+        """Add a publish instance to the current project.
+
+        Args:
+            instance_data (dict): Publish instance to add.
+
+        """
+        data = self.get_ayon_data()
+        publish_instances = self.get_publish_instances()
+        publish_instances.append(instance_data)
+        data[EQUALIZER_INSTANCES_KEY] = publish_instances
+
+        self.update_ayon_data(data)
+
+    def update_publish_instance(
+            self,
+            instance: CreatedInstance,
+            data: dict,
+    ) -> None:
+        """Update a publish instance in the current project.
+
+        Args:
+            instance (CreatedInstance): Publish instance to update.
+            data (dict): Data to update.
+
+        """
+        ayon_data = self.get_ayon_data()
+        publish_instances = self.get_publish_instances()
+        for idx, publish_instance in enumerate(publish_instances):
+            if publish_instance["instance_id"] == instance.id:
+                publish_instances[idx] = data
+                break
+        ayon_data[EQUALIZER_INSTANCES_KEY] = publish_instances
+
+        self.update_ayon_data(ayon_data)
+
+    def write_publish_instances(
+            self, instances: list[dict]) -> None:
+        """Write publish instances to the current project."""
+        ayon_data = self.get_ayon_data()
+        ayon_data[EQUALIZER_INSTANCES_KEY] = instances
+        self.update_ayon_data(ayon_data)
+
+    def remove_publish_instance(self, instance: CreatedInstance) -> None:
```

Comment:
And again, instance is used only to get it's id, you can expect only id.
```suggestion
    def remove_create_instance(self, instance: CreatedInstance) -> None:
```